app/components/progress_badge_component.rb

```diff
@@ -19,11 +19,17 @@ def circumference
   end
 
   def badge
-    course_badges.fetch(course.title, 'icons/odin-icon.svg')
+    @course.badge_uri || fallback_icon
   end
 
   def borderless_badge
-    borderless_badges.fetch(course.title, 'icons/odin-icon.svg')
+    badge_uri = @course.badge_uri.split('.').first
+
+    "badges/#{badge_uri}-borderless" || fallback_icon
+  end
+
+  def fallback_icon
```

Comment:
Looking over this again, this should probably be a constant. Thoughts? Probably not a huge deal.

Reply:
Method looks good to me too 👍